glyphs-reader/src/font.rs

```diff
@@ -632,6 +682,34 @@ fn v2_to_v3_name(properties: &mut Vec<RawName>, v2_prop: &Option<String>, v3_nam
     }
 }
 
+/// Convert to axesValues, handy for a master or instance
+fn v2_to_v3_axis_values(
+    axes: &Vec<Axis>,
+    other_stuff: &mut BTreeMap<String, Plist>,
+) -> Result<Vec<OrderedFloat<f64>>, Error> {
+    let mut axis_values = Vec::new();
+    for idx in 0..axes.len() {
+        // Per https://github.com/googlefonts/fontmake-rs/pull/42#pullrequestreview-1211619812
+        // the field to use is based on the order in axes NOT the tag.
+        // That is, whatever the first axis is, it's value is in the weightValue field. Long sigh.
+        let (field_names, default_value) = glyphs_v2_field_name_and_default(idx)?;
+        let value = field_names
+            .iter()
+            .filter_map(|field_name| other_stuff.remove(*field_name))
+            .next()
+            .unwrap_or_else(|| Plist::Float(default_value.into()))
+            .as_f64()
```

Comment:
Looking at this more closely, the flow is a little funny. We want to return an error if there *is* a value in the plist for this field name, but it is the wrong type (not a number?) but we want to use a default value if there is no plist value for this field name?

If that is true, I would prefer to write this as something like,

```rust
        let value = field_names
            .iter()
            .find_map(|field_name| {
                other_stuff.remove(*field_name).map(|x| {
                    x.as_f64().ok_or_else(|| {
                        Error::StructuralError(format!(
                            "Invalid value {x:?} for field {field_name}"
                        ))
                    })
                })
            })
            .unwrap_or(Ok(default_value))?;
        axis_values.push(value.into());
```

the basic idea being to generate the error where it occurs, instead of unwrapping with a known good value and validating at the end?


This is also just for illustrative purposes, no change required.

Reply:
In re-reading I think I will change to issue a warning if it's a weird value